integration/commands_administration_compat_test.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"errors"
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+
+	"github.com/FerretDB/FerretDB/integration/setup"
+	"github.com/FerretDB/FerretDB/integration/shareddata"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+)
+
+func TestCommandsAdministrationCompatCollStatsWithScale(t *testing.T) {
+	t.Parallel()
+
+	s := setup.SetupCompatWithOpts(t, &setup.SetupCompatOpts{
+		Providers:                []shareddata.Provider{shareddata.DocumentsDocuments},
+		AddNonExistentCollection: true,
+	})
+
+	ctx, targetCollection, compatCollection := s.Ctx, s.TargetCollections[0], s.CompatCollections[0]
+
+	for name, tc := range map[string]struct { //nolint:vet // for readability
+		scale      any
+		resultType compatTestCaseResultType
+		altMessage string
+	}{
+		"scaleOne":      {scale: int32(1)},
+		"scaleBig":      {scale: int64(1000)},
+		"scaleMaxInt":   {scale: math.MaxInt64},
+		"scaleZero":     {scale: int32(0), resultType: emptyResult},
+		"scaleNegative": {scale: int32(-100), resultType: emptyResult},
+		"scaleFloat":    {scale: 2.8},
+		"scaleMinFloat": {scale: -math.MaxFloat64, resultType: emptyResult},
+		"scaleMaxFloat": {scale: math.MaxFloat64},
```

Comment:
This test is unhappy in CI

```
2023-04-12T23:36:01.2508220Z     --- FAIL: TestCommandsAdministrationCompatDBStatsWithScale/scaleMaxFloat (0.02s)
2023-04-12T23:36:01.2508482Z         commands_administration_compat_test.go:136: 
2023-04-12T23:36:01.2509051Z             	Error Trace:	/home/runner/work/FerretDB/FerretDB/integration/commands_administration_compat_test.go:166
2023-04-12T23:36:01.2509311Z             	Error:      	Received unexpected error:
2023-04-12T23:36:01.2509609Z             	            	scale has to be > 0
2023-04-12T23:36:01.2510060Z             	Test:       	TestCommandsAdministrationCompatDBStatsWithScale/scaleMaxFloat
```

Reply:
Oh, that's interesting :) It passes for me locally, maybe because of a different architecture.
I will remove this case for now (like I did with other cases), because the error message (and error code 0) that Mongo returns, doesn't make any sense 😅

Reply:
Please never remove test cases for found problems to make tests pass

Reply:
Ok, I'll add diff.md changes and dance tests then.

Reply:
Or we could just skip them with an issue number

Reply:
But the current issue is _the_ issue number to have a proper fix :)

Reply:
I believe we have an issue for architecture-specific float64 problems

Reply:
#491. Please make it pass on CI and skip it on arm64. Search for `runtime.GOARCH == "arm64"`

Reply:
Well, this problem is also about Mongo returning 0 error code instead of `BadValue`.

Reply:
That's not new too. See `getParameter` implementation, for example

Reply:
Added a diff doc and a PR: https://github.com/FerretDB/dance/pull/407